omg/core/views.py

```diff
@@ -145,6 +150,35 @@ def update(self, delta_time):
     def _on_projectile_shot(self, event: ProjectileShotEvent):
         self.projectiles.append(event.projectile)
 
+    def _check_collision_between_player_and_pickupable(self):
+        collided_sprites: list[Pickupable] = arcade.check_for_collision_with_list(
+            self.player, self.pickupables
+        )
+        return collided_sprites
+
+    def _draw_pickup_button(self):
+        collided_pickupables = self._check_collision_between_player_and_pickupable()
+        if len(collided_pickupables) >= 1:
+            for pickupable in collided_pickupables:
+                # Get button image
+                button = arcade.Sprite(
+                    self.pickup_button_image,
+                    scale=self.pickup_button_image_scale
+                )
+
+                # Calculate directional vector between player and pickupable
+                diff_x: float = pickupable.center_x - self.player.center_x
+                diff_y: float = pickupable.center_y - self.player.center_y
+
+                # Place button image at the mirror reflection of player wrt pickupable
+                button.center_x = pickupable.center_x + diff_x
+                button.center_y = pickupable.center_y + diff_y
+
+                # Draw the button image
+                button.draw()
+        else:
+            return
```

Comment:
this part includes just more than drawing. As discussed, I believe a different approach would make it redundant to do all the collision calculations,